spec/models/user_spec.rb

```diff
@@ -30,13 +30,13 @@
     end
   end
 
-  describe '.as_json' do
+  describe 'data' do
     context 'when array of instances converting to json' do
       let(:project) { create(:project) }
       let(:user) { create(:user, project: project) }
 
       it 'returns an array of objects as json' do
-        as_json = JSON.parse(User.as_json([user])).first
+        as_json = JSON.parse(user.data.to_json)
```

Comment:
You do not need to user `to_json` and after parse this json.
`user_data = user.data`